test/rubygems/test_gem.rb

```diff
@@ -777,6 +777,22 @@ def test_self_ruby_api_version
     RbConfig::CONFIG['ruby_version'] = orig_ruby_version
   end
 
+  def test_self_env_requirement
+    old_env = ENV.to_hash
+
+    ENV.clear
```

Comment:
RubyGems tests are supposed to run with `$SAFE = 1` but only ruby 1.8.7 does this.  You cannot wipe ENV with `$SAFE = 1` so you should instead clear these in setup and teardown (missing) [up here](https://github.com/segiddins/rubygems/blob/seg-env-req-tests/test/rubygems/test_gem.rb#L19).